TrafficSimulator/Assets/Utilities/CarSpawner.cs

```diff
@@ -142,32 +148,72 @@ private void SpawnCars()
             {
                 // Calculate the number of cars to spawn
                 int carsToSpawn = _mode == SpawnMode.Total ? Mathf.CeilToInt(_ratios[i] * TotalCars) : Mathf.CeilToInt(_maxCarsPerLane[i] * LaneCarRatio);
-
                 // Return if there are no cars to spawn
                 if (carsToSpawn == 0)
                     return;
 
                 // Calculate the offset
                 _laneNodeCurrent = _lanes[i].StartNode;
 
-                // Spawn cars
-                for (int j = 0; j < carsToSpawn; j++)
+                List<float> sections = DivideLaneToSections(_lanes[i]);
+                List<float> sectionRatios = CalculateSectionRatios(sections);
+                for (int j = 0; j < sections.Count; j++)
                 {
-                    // Check if max cars have been spawned
-                    if (_mode == SpawnMode.Total && _carCounter >= TotalCars)
-                        return;
+                    // Calculate the number of cars to spawn
+                    int carsToSpawnSection = Mathf.CeilToInt(carsToSpawn * sectionRatios[j]);
+                    for (int k = 0; k < carsToSpawnSection; k++)
+                    {
+                        // Check if max cars have been spawned
+                        if (_mode == SpawnMode.Total && _carCounter >= TotalCars)
+                            return;
+
+                        // Spawn car
+                        if(!_laneNodeCurrent.RoadNode.IsIntersection() && _laneNodeCurrent.RoadNode.Type != RoadNodeType.JunctionEdge && _laneNodeCurrent != null && !_laneNodeCurrent.HasVehicle() && _laneNodeCurrent.Next != null && _laneNodeCurrent.Next.Position != _laneNodeCurrent.Position)
```

Comment:
Very long line with alot of logic, should probably split it into multiple bools to increase readability